packages/devtools_app/lib/src/shared/ui/filter.dart

```diff
@@ -10,68 +11,158 @@ import '../primitives/auto_dispose.dart';
 import '../primitives/utils.dart';
 import '../theme.dart';
 
-// TODO(kenz): consider breaking this up flat data filtering and tree data
+// TODO(kenz): consider breaking this up for flat data filtering and tree data
 // filtering.
-mixin FilterControllerMixin<T> {
+
+/// Mixin to add to feature controllers that need to support filtering data and
+/// storing the state of those filters.
+///
+/// To use this mixin, you must implement [filterData] as well as either or both
+/// [createToggleFilters] and [createQueryFilterArgs].
+///
+/// Classes mixing in [FilterControllerMixin] must also extend
+/// [DisposableController] and mixin [AutoDisposeControllerMixin], and a class
+/// can subscribe to updates to the active filter by calling
+/// [subscribeToFilterChanges].
+mixin FilterControllerMixin<T> implements AutoDisposeControllerMixin {
```

Comment:
If this can only be applied to classes extending `DisposableController`, we should enforce that:

```dart
mixin FilterControllerMixin<T> on DisposableController implements AutoDisposeControllerMixin {...}
```